### ISSUE TITLE
Lost socket after disconnect/reconnect

### ISSUE THREAD

**Issue by hunterjm** [OpenXbox/xbox-smartglass-core-python#7]
I think I found another issue.   Not sure where exactly it is, but fairly easy to replicate in both my PR branch, and the master branch.

Steps to reporduce:
1) Load TUI with console already on.
2) Connect to console with "c"
3) Disconnect from console with "d"
4) Re-connect to console with "c"
5) Wait for first `_heartbeat`

Result:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/gevent/greenlet.py", line 536, in run                                                                                                                                                         
    result = self._run(*self.args, **self.kwargs)                                                                                                                                                                                            
  File "/Users/jason/xbox-smartglass-core-python/xbox/sg/protocol.py", line 265, in _heartbeat                                                                                                                                               
    self.ack([], [], ServiceChannel.Core, need_ack=True)                                                                                                                                                                                    
  File "/Users/jason/xbox-smartglass-core-python/xbox/sg/protocol.py", line 498, in ack                                                                                                                                                     
    self.send_message(msg, channel=channel, blocking=need_ack)                                                                                                                                                                              
  File "/Users/jason/xbox-smartglass-core-python/xbox/sg/protocol.py", line 162, in send_message                                                                                                                                            
    self.socket.sendto(data, (addr, PORT))                                                                                                                                                                                                  
AttributeError: 'CoreProtocol' object has no attribute 'socket'                                                                                                                                                                             
Sat Jun 16 11:51:33 2018 <Greenlet at 0x10a035930: <bound method CoreProtocol._heartbeat of <CoreProtocol at 0x109fb8550 address=0.0.0.0:61568>>> failed with AttributeError
```

**Comment by Schamper** [comment]
Identified and fixed. Found other bug in the meantime though.

**Comment by Schamper** [comment]
Will be fixed with #6
